File: javascript/bob/bob.js
class Bob {
  hey(phrase) {
    // console.log(phrase, phrase.toUpperCase(), phrase === phrase.toUpperCase());
    // if (phrase.match(/^[A-Z0-9?!, ]+[A-Z]{2}[A-Z0-9?!, ]+$/)) {
    if (phrase.match(/[A-Z]{2}/) && phrase === phrase.toUpperCase()) {
      return 'Whoa, chill out!';
    } else if (phrase.endsWith('?')) {
      return 'Sure.';
    } else if (phrase.trim() === '') {
      return 'Fine. Be that way!';
    } else {
      return 'Whatever.';
    }
  }
}

module.exports = Bob;

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.match? input, ~r/.*\?$/ -> "Sure."
      String.upcase(input) == input && String.downcase(input) != input -> "Whoa, chill out!"
      String.match? input, ~r/^\s*$/ -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end
end

IO.puts Bob.hey("УХОДИ")
